index.js

```diff
@@ -125,7 +125,10 @@ Discovery.prototype.destroy = function (cb) {
     self.tracker.removeListener('peer', self._onTrackerPeer)
     self.tracker.removeListener('update', self._onTrackerAnnounce)
     tasks.push(function (cb) {
-      self.tracker.destroy(cb)
+      process.nextTick(function () {
+        self.tracker.destroy(cb)
```

Comment:
Great! Sorry I was not more help. 

Sometime later I would love to know your WS implementation. I am almost done with my TCP protocol implementation and I want my WS to be compatible with yours. I'm assuming you just send data like BEP09 metadata.. a bencoded dictionary followed by the data.